src/java.base/share/classes/jdk/internal/reflect/ReflectionFactory.java

```diff
@@ -552,6 +552,11 @@ static boolean useOldSerializableConstructor() {
         return config().useOldSerializableConstructor;
     }
 
+    public static boolean useLegacyProxyImpl() {
+        var config = config();
+        return config.useLegacyProxyImpl && !config.useOldSerializableConstructor;
```

Comment:
```suggestion
        return config.useLegacyProxyImpl || config.useOldSerializableConstructor;
```

Reply:
This site can actually simply be `config.useLegacyProxyImpl` as it's initialized in `loadConfig`.